design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -79,55 +80,40 @@ The `id` will be a `base64_encode(increment_id)` which in future can be replaced
 
 ### Order Item
 
-The order items will be presented as separate interface which will have multiple implementations for invoice, shipment and credit memo types.
-
 ```graphql
-@doc("Interface to reprent order/invoice/shipment/credit memo items")
-interface SalesItemInterface {
+@doc("Order item")
+type OrderItemInterface {
+    id: ID! @doc("Order item unique identifier") #base64encode(orderItemId)
     product_name: String @doc("name of the base product")
     product_sku: String! @doc("SKU of the base product")
-    product_url: String @doc("URL of the base product")
+    product_url_key: String @doc("URL key of the base product")
+    product_type: String @doc("Type of product (e.g. simple, configurable, bundle)")
+    status: String @doc("the status of order item")
     product_sale_price: Money! @doc("sale price for the base product including selected options")
     discounts: [Discount] @doc("final discount information for the base product including discounts on options")
-    parent_product_name: String @doc("name of parent product like configurable or bundle")
-    parent_product_sku: String @doc("SKU of parent product like configurable or bundle")
-    parent_product_url: String @doc("URL of parent product in the catalog")
-    selected_options: [SalesItemSelectedOption] @doc("selected options for the base product. for e.g color, size etc.")
-    entered_options: [SalesItemEnteredOption] @doc("entered option for the base product. for e.g logo image etc.")
-}
-
-@doc("Represents sales item selected options")
-type SalesItemSelectedOption {
-    id: ID! @doc("ID of the option")
-    label: String! @doc("name of the option")
-    value_labels: [String]! @doc("list of option value labels")
-}
-
-@doc("Represents sales item entered options")
-type SalesItemEnteredOption {
-    id: ID! @doc("ID of the option")
-    label: String! @doc("name of the option")
-    value: String! @doc("value of the option")
-}
-```
-
-The `id` will be a `base64_encode(option_id)` which in future can be replaced by UUID.
-
-The `SalesItemInterface` will be implemented by the following types:
-
-```graphql
-@doc("Order Product implementation of OrderProductInterface")
-type OrderItem implements SalesItemInterface {
+    selected_options: [OrderItemOption] @doc("selected options for the base product. for e.g color, size etc.")
+    entered_options: [OrderItemOption] @doc("entered option for the base product. for e.g logo image etc.")
     quantity_ordered: Float @doc("number of items")
     quantity_shipped: Float @doc("number of shipped items")
     quantity_refunded: Float @doc("number of refunded items")
     quantity_invoiced: Float @doc("number of invoiced items")
     quantity_canceled: Float @doc("number of cancelled items")
     quantity_returned: Float @doc("number of returned items")
-    status: String @doc("the status of order item")
 }
-```
 
+type OrderItem implements OrderItemInterface {
```

Comment:
```
type OrderItem implements OrderItemInterface {
}
```

This is invalid according to the GraphQL spec

![image](https://user-images.githubusercontent.com/5233399/83554981-33a92800-a4d3-11ea-9fd5-c450439f3b17.png)

Source: http://spec.graphql.org/June2018/#sec-Interfaces

Reply:
The interface does define one or more fields. The type that implements the interface does not add any additional fields. That statement isn't 100% clear to me so maybe I got it wrong

Reply:
Sorry, meant to link the validation for Object types.

![image](https://user-images.githubusercontent.com/5233399/83557810-e8dddf00-a4d7-11ea-8fca-fe898333837e.png)

http://spec.graphql.org/June2018/#sec-Objects

Have you tested this works with webonyx? If so it sounds like there's a spec-compliancy issue.

Reply:
Here's how the GraphQL reference implementation from the GraphQL foundation does it:

![image](https://user-images.githubusercontent.com/5233399/83558869-46266000-a4d9-11ea-985d-0c2533bd9c09.png)

Note that validation is done in the same order as the spec. So checks for fields are done _before_ checking if the interface is respected.

Reply:
(I bring this up because we won't be able to implement this schema in the new GraphQL server)

Reply:
The schema stitching will copy the interface fields into the type so we don't need to redeclare all interface fields on the type. This is the approach used in other schema.

Reply:
@paliarush also pointed this out to me in another PR 👍. Thanks for clarifying